build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ResolveAllDependencies.java

```diff
@@ -37,18 +36,32 @@ public abstract class ResolveAllDependencies extends DefaultTask {
     protected abstract JavaToolchainService getJavaToolchainService();
 
     private final ObjectFactory objectFactory;
+    private final ProviderFactory providerFactory;
 
     private Collection<Configuration> configs;
 
     @Inject
-    public ResolveAllDependencies(ObjectFactory objectFactory) {
+    public ResolveAllDependencies(ObjectFactory objectFactory, ProviderFactory providerFactory) {
         this.objectFactory = objectFactory;
+        this.providerFactory = providerFactory;
     }
 
     @InputFiles
     public FileCollection getResolvedArtifacts() {
         return objectFactory.fileCollection()
-            .from(configs.stream().filter(ResolveAllDependencies::canBeResolved).collect(Collectors.toList()));
+            .from(configs.stream()
+                .filter(ResolveAllDependencies::canBeResolved)
+                .map(c -> {
+                    // Make a copy of the configuration, omitting file collection dependencies to avoid building project artifacts
+                    Configuration copy = c.copyRecursive(d -> d instanceof FileCollectionDependency == false);
+                    copy.setCanBeConsumed(false);
+                    return copy;
+                })
+                // Include only module dependencies, ignoring things like project dependencies so we don't unnecessarily build stuff
+                .map(c -> c.getIncoming().artifactView(v -> v.lenient(true).componentFilter(i -> i instanceof ModuleComponentIdentifier)))
```

Comment:
We also make the artifact view lenient as some of these file collection dependencies are "required" causing resolution failures but we don't care in this context.